packages/core/src/browser/connection-status-service.ts

```diff
@@ -230,33 +229,22 @@ export class ApplicationConnectionStatusContribution extends DefaultFrontendAppl
         }
     }
 
-    protected getOrCreateDialog(content: string): ConnectionStatusDialog {
-        if (this.dialog === undefined) {
-            this.dialog = new ConnectionStatusDialog({
-                title: 'Not connected',
-                content
-            });
-        }
-        return this.dialog;
-    }
+    private statusbarId = 'connection-status';
 
     protected handleOnline() {
-        const message = 'Successfully reconnected to the backend.';
-        this.logger.info(message);
-        if (this.dialog !== undefined) {
-            this.dialog.dispose();
-            this.dialog = undefined;
-        }
+        this.statusBar.setBackgroundColor(undefined);
+        this.statusBar.removeElement(this.statusbarId);
     }
 
     protected handleOffline() {
-        if (this.dialog === undefined) {
-            const message = 'The application connection to the backend is lost. Attempting to reconnect...';
-            this.logger.error(message);
-            this.getOrCreateDialog(message).open();
-        }
+        this.statusBar.setBackgroundColor('var(--theia-warn-color0)');
+        this.statusBar.setElement(this.statusbarId, {
+            alignment: StatusBarAlignment.LEFT,
+            text: 'OFFLINE!',
```

Comment:
I do not like this that much. I believe an `OFFLINE` or `Offline` would be sufficient. But I am fine with this too.